From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EpicKnarvik97 <kristian.knarvik@knett.no>
Date: Sat, 5 Mar 2022 20:58:46 +0100
Subject: [PATCH] Expose furnace minecart push values

Adds methods for getting and setting a furnace minecart's push values

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftMinecartFurnace.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftMinecartFurnace.java
index 49ceb730b6e68b7c70799fca80dc32da4a12c545..b8378d5f3c2a08ab565dcb8cb200822b581c7dba 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftMinecartFurnace.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftMinecartFurnace.java
@@ -28,6 +28,28 @@ public class CraftMinecartFurnace extends CraftMinecart implements PoweredMineca
         this.getHandle().fuel = fuel;
     }
 
+    // Paper start
+    @Override
+    public double getPushX() {
+        return getHandle().xPush;
+    }
+
+    @Override
+    public double getPushZ() {
+        return getHandle().zPush;
+    }
+
+    @Override
+    public void setPushX(double xPush) {
+        getHandle().xPush = xPush;
+    }
+
+    @Override
+    public void setPushZ(double zPush) {
+        getHandle().zPush = zPush;
+    }
+    // Paper end
+
     @Override
     public String toString() {
         return "CraftMinecartFurnace";
